pkg/handler/handler.go

```diff
@@ -55,13 +56,40 @@ type OpenAPIService struct {
 
 	lastModified time.Time
 
-	specBytes []byte
-	specPb    []byte
-	specPbGz  []byte
+	jsonCache  cache
+	protoCache cache
+}
+
+type cache struct {
+	BuildCache func() ([]byte, error)
+	once       sync.Once
+	bytes      []byte
+	etag       string
+	err        error
```

Comment:
The `err` field is a bit confusing. Readers may wonder of what process was this error produced. We can make it local. See suggestion below.

Reply:
Makes sense, I think only `cache.Get` was using that field. Fixed, thanks!

Reply:
Hmm, I'm not sure this handles the case of multiple failing calls to cache.Get(). The first call will return the error, but the second will return `nil`. https://play.golang.org/p/3RgPZBfAvCH outputs: `[SECOND CALL] expected non-nil err from cache.Get()`

Reply:
So...should we retry or do we want to return the same error every time?

Reply:
I think saving `err` and returning it again is correct in the context of what we're trying to do here - an error that occurs during marshaling (which is what would get recorded here) is very likely to occur again IMO if we retry/call out to `json.Marshal` again, so it'd be best to wait for another call to `UpdateSpec` that should result in a fresh new spec to marshal.

I think in regard to [this discussion](https://github.com/kubernetes/kube-openapi/pull/251#discussion_r740640134), saving `err` makes sense as well - if we didn't save `err`, the first call would result in an HTTP 503, but any following calls would result in an HTTP 200 with an empty body.

Reply:
I see...you can drop the latest commit to revert back